lua/nvim_lsp/rust_analyzer.lua

```diff
@@ -0,0 +1,49 @@
+-- Author: Heyward Fann <fannheyward@gmail.com>
+-- Description: rust-analyzer for nvim-lsp
+
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local vim = vim
+local lsp = vim.lsp
+
+skeleton.rust_analyzer = {
+  default_config = {
+    cmd = {"ra_lsp_server"};
+    filetypes = {"rust"};
+    root_dir = util.root_pattern("Cargo.toml", ".git");
+    log_level = lsp.protocol.MessageType.Warning;
+    settings = {};
+  };
+  commands = {
+  };
+  -- on_new_config = function(new_config) end;
+  -- on_attach = function(client, bufnr) end;
+  docs = {
+    description = [[
+https://rust-analyzer.github.io
+
+Bringing great IDE experience to the Rust programming language.
+
+the following settings (with default options) are supported:
+```lua
+settings = {
+  ["rust-analyzer"] = {
+    trace = { server = "verbose"; };
+    excludeGlobs = {};
```

Comment:
There are many more flags available:

rust-analyzer.highlightingOn: enables experimental syntax highlighting
rust-analyzer.enableEnhancedTyping: by default, rust-analyzer intercepts Enter key to make it easier to continue comments. Note that it may conflict with VIM emulation plugin.
rust-analyzer.raLspServerPath: path to ra_lsp_server executable
rust-analyzer.enableCargoWatchOnStartup: prompt to install & enable cargo watch for live error highlighting (note, this does not use rust-analyzer)
rust-analyzer.excludeGlobs: a list of glob-patterns for exclusion (see globset docs for syntax). Note: glob patterns are applied to all Cargo packages and a rooted at a package root. This is not very intuitive and a limitation of a current implementation.
rust-analyzer.useClientWatching: use client provided file watching instead of notify watching.
rust-analyzer.cargo-watch.command: cargo-watch command. (e.g: clippy will run as cargo watch -x clippy )
rust-analyzer.cargo-watch.arguments: cargo-watch check arguments. (e.g: --features="shumway,pdf" will run as cargo watch -x "check --features="shumway,pdf"" )
rust-analyzer.cargo-watch.ignore: list of patterns for cargo-watch to ignore (will be passed as --ignore)
rust-analyzer.trace.server: enables internal logging
rust-analyzer.trace.cargo-watch: enables cargo-watch logging
RUST_SRC_PATH: environment variable that overwrites the sysroot
rust-analyzer.featureFlags -- a JSON object to tweak fine-grained behavior: